### PR TITLE
Thchang/ issue_1319

### LINKED ISSUE #1319: world coordinate offset in the spatial profiler
Please refer to the following video and focus on RA=13:30:00.00. The cursor stays as RA=13:30:00.00 while zooming and the red vertical line should stay at RA=13:30:00.00. However, there is offset at different zoom levels. 
In addition, at high zoom level, there seems to be an 1-pixel offset (due to the legacy 0-index/1-index issue?).

https://user-images.githubusercontent.com/20819712/108473853-8a58cb00-72c9-11eb-8e7e-b01c65c67d3c.mp4

### PR DESCRIPTION
**Pull request by TienHao**
**Description**
fixed #1319 by adding needed digits after decimal point(depending on cursor info precision and wcs value) in the upper wcs axis in spatial profilers.

**Checklist**
For linked issues (if there are):
- [x] assignee and label added
- [x] ZenHub issue connection, board status, and estimate updated

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] e2e test passing / ~corresponding fix added~
- [x] changelog updated
- [x] no protobuf update needed
- [x] `BackendService` unchanged

### CODE REVIEW

**Review by kswang1029 — COMMENTED** [review]
Overall this works nicely. 👍 No regression is observed from e2e tests. 

Few minor issues to consider but not all of them need to be addressed:
1. [required] when the 2nd tick values are in channel, please use plain value format (ie remove ,). Note this exists in dev so it is not due to this PR.
Screen Shot 2022-10-17 at 15 14 07

2. [best effort] in some cases, the tick values are too crowded. If possible, in this case we need to skip some.
Screen Shot 2022-10-17 at 15 30 42

3. [can ignore] at extreme zoom levels, we start to see tick values are repeating.
Screen Shot 2022-10-17 at 15 39 51

Code review may start.

**Review by YuHsuan-Hwang — APPROVED** [review]

**Comment by TienHao** [comment]
> 1. [required] when the 2nd tick values are in channel, please use plain value format (ie remove ,). Note this exists in dev so it is not due to this PR.
> 2. [best effort] in some cases, the tick values are too crowded. If possible, in this case we need to skip some.
> 3. [can ignore] at extreme zoom levels, we start to see tick values are repeating.

@kswang1029 : 
I have fixed the minor issues 1 & 2 first and kept the third one remained.

**Review by kswang1029 — CHANGES_REQUESTED** [review]
@TienHao thanks for addressing the value format and value overlapping issues. 👍 
Some e2e tests failed due to the top axis. Please hide the top axis for the spatial profiler widget if the region type is line or polyline. Please also hide the top axis for the spectral profiler.
Screen Shot 2022-10-28 at 11 00 45
Screen Shot 2022-10-28 at 11 00 31

**Review by kswang1029 — APPROVED** [review]
this works well now. no regression is observed from e2e tests. 👍

**Review by YuHsuan-Hwang — APPROVED** [review]

File: src/components/SpatialProfiler/SpatialProfilerComponent.tsx [hunk 1]
```diff
@@ -440,13 +466,10 @@ export class SpatialProfilerComponent extends React.Component<WidgetProps> {
...
                 // handle the value when cursor is in profiler
-                const nearest = binarySearchByX(this.plotData.values, this.widgetStore.cursorX);
-                if (nearest?.point) {
-                    const pixelPoint = isXCoordinate ? {x: nearest.point.x, y: this.profileStore.y} : {x: this.profileStore.x, y: nearest.point.x};
-                    const cursorInfo = this.frame.getCursorInfo(pixelPoint);
-                    const wcsLabel = cursorInfo?.infoWCS && !this.lineAxis ? `WCS: ${isXCoordinate ? cursorInfo.infoWCS.x : cursorInfo.infoWCS.y}, ` : "";
-                    const xLabel = this.lineAxis ? `${this.lineAxis.label}: ${formattedExponential(nearest.point.x, 5)} ${this.lineAxis.unit ?? ""}, ` : `Image: ${nearest.point.x} px, `;
-                    const valueLabel = `${nearest.point.y !== undefined ? formattedExponential(nearest.point.y, 5) : ""}`;
+                if (this.pointInfo && this.nearestCursorPoint) {
+                    const wcsLabel = this.pointInfo?.infoWCS && !this.lineAxis ? `WCS: ${isXCoordinate ? this.pointInfo.infoWCS.x : this.pointInfo.infoWCS.y}, ` : "";
```
Comment (crocka) [inline src/components/SpatialProfiler/SpatialProfilerComponent.tsx]: Can we replace isXCoordinate with this.widgetStore.isXProfile?

**Review by crocka — APPROVED** [review]

**Review by veggiesaurus — APPROVED** [review]

**Pull request merged**